### ISSUE TITLE
Thinking of using wiktionary.org as source for Free Dictionary API. What are your thoughts? 

### ISSUE THREAD

**Issue by meetDeveloper** [meetDeveloper/freeDictionaryAPI#102]

**Comment by rrbadam** [comment]
Fine with it

**Comment by queleok** [comment]
Would it be an additional source, or a replacement for the current one?

**Comment by meetDeveloper** [comment]
@queleok This will be an replacement.

**Comment by meetDeveloper** [comment]
Wikitionary uses following as sources: https://en.wiktionary.org/wiki/Wiktionary:Public_domain_sources

**Comment by queleok** [comment]
I see, thank you, @meetDeveloper . I've tried checking the dictionary of 100k most frequently used English words according to the Wiktionary, there were some odd items like clearly French or German words. Of course, it's not a showstopper for the idea, so I guess it should be fine :)

**Comment by phibr0** [comment]
When will this change happen?

**Comment by ichmoimeyo** [comment]
HI

Have you checked out ...

https://yandex.com/dev/dictionary/
https://yandex.com/legal/dictionary_api/ **FREE**

http://www.collinslanguage.com/collins-api/ **FREE -> 5000 p/m | £75p/m -> 30000 p/m**

**Comment by meetDeveloper** [comment]
@ichmoimeyo They support English which I can get from Wiktionary.

**Comment by ichmoimeyo** [comment]
@meetDeveloper 
https://yandex.com/dev/dictionary/doc/dg/reference/getLangs.html **[with russian emphasis]**
http://www.collinslanguage.com/collins-api/ **[Bilingual or monolingual/bilingual]**

**Comment by ms-dc** [comment]
any eta to non english language coming back? portuguese in particular

**Comment by piever** [comment]
> The only downside I see is that the API will be exclusively for English language as for other languages as Wiktionary lags behind, but you can expect other languages to be added as soon as possible. Given that majority of users of this API are English speaking users, this change will affect relatively fewer users.

I confess I would love it if more languages were available. At a cursory look, it seems as if Wiktionary is reasonably good in other languages as well. Is the problem that the wiktionary API is inconsistent across languages?

Otherwise, I imagine it would cost relatively little to just support other languages as well (even if it turns that Wiktionary lags behind a bit on those, I guess it's still better than not supporting them at all).

**Comment by sbeve72** [comment]
I can't complain as I pay 0 bucks to use this api but this will break [my app](www.github.com/sbeve72/JADA). Brazilian Portuguese is already broken :'(

**Comment by OultimoCoder** [comment]
Yeah I would really prefer the change didn't happen. I am creating a card for a language flashcard apps and this is super useful for dictionary definitions in a large variety of languages.

**Comment by arshsaxena** [comment]
Is this migration completed?

**Comment by meetDeveloper** [comment]
@arshsaxena @ac-lap @sbeve72 @OultimoCoder @piever This change will happen today. I hope to bring other languages soon but parsing Wiktionary for other languages is a bit difficult and will take some time.

**Comment by meetDeveloper** [comment]
This migration is completed. Now the API is only for English and using Wiktionary as source also I am sending in response sourceUrls from which the definitions have been parsed and the licenses.

**Comment by 1over137** [comment]
I think this will make this API completely useless, as English Wiktionary already has its own API.

**Comment by piever** [comment]
Thanks for the notification!

I imagine (correct me if I'm wrong) that at this point, this project is primarily a "conversion layer" between the Wiktionary API and the free dictionary API. Personally, I think that it is valuable, in that it was pretty straightforward for me to work with this API, but it looks like working directly with the Wiktionary API would have been harder.

Sorry about the slight off-topic, but I was wondering... Given that (from what I see in the README) it is costly to keep up servers that provide this API, would it make sense to package this as a plain module that one could load (say from CDN) and use it directly (to convert the result of calls to the Wiktionary API)?

**Comment by ngocsangyem** [comment]
Does the source code need to be updated @meetDeveloper ?

**Comment by meetDeveloper** [comment]
@ngocsangyem yes it needs to be updated.

**Comment by meetDeveloper** [comment]
@ngocsangyem yes it needs to be updated.

**Comment by ngocsangyem** [comment]
It is very nice If you update the new source code soon. I need to make some customs for the API to host my app. @meetDeveloper

**Comment by meetDeveloper** [comment]
@piever Yes you are right, Wiktionary exposes dumps that anyone can parse and use it in their own work. I was also thinking to package this API in a plain module to avoid server costs, this is something I am exploring and will likely start work in few weeks.

**Comment by Etiene** [comment]
> Yeah I would really prefer the change didn't happen. I am using this API for a language flashcard app and this is super useful for dictionary definitions in a large variety of languages.

I agree!

**Comment by spacelover1** [comment]
I want to use Oxford dictionary API and have trouble passing API key and ID. I get response using postman but in code it doesn't pass them. Anyone used this API?

**Comment by abhagsain** [comment]
> This migration is completed. Now the API is only for English and using Wiktionary as source also I am sending in response sourceUrls from which the definitions have been parsed and the licenses.

It'd be great if you merge your latest changes to master @meetDeveloper

**Comment by alexguillamon** [comment]
This is a useful project but others would like to run the server locally and or further make changes can you please push the latest changes to master? @meetDeveloper

**Comment by AndrewDiMola** [comment]
Has anyone created an adapted list of words that can be retrieved after the migration change? I used to use this file list as a source of truth, but because of the migration, a lot of words no longer retrieve results. 

https://github.com/meetDeveloper/freeDictionaryAPI/blob/master/meta/wordList/english.txt

**Comment by KlavsKlavsen** [comment]
can anyone point to how to add a new language from wictionary to this? its really a great idea to use wictionary as source - hoping I can figure out how to add Hindi - so I can use this (and possibly improve - by using wictionary)

**Comment by PSYCHEER** [comment]
I think it's going to support english only in the near future...
As no other language code doesn't work...
